patroni/dcs/etcd.py

```diff
@@ -264,8 +248,11 @@ def _get_machines_cache_from_srv(self, srv):
 
     def _get_machines_cache_from_dns(self, host, port):
         """One host might be resolved into multiple ip addresses. We will make list out of it"""
-        ret = ['{0}://{1}:{2}'.format(self.protocol, ip, port) for ip in set(self._dns_resolver.resolve(host))]
-        return list(set(ret)) if ret else ['{0}://{1}:{2}'.format(self.protocol, host, port)]
+        if self.protocol == 'http':
+            ret = ['{0}://{1}:{2}'.format(self.protocol, *r[4]) for r in self._dns_resolver.resolve(host, port)]
```

Comment:
Actually, is this method still correct? Wouldn't  _get_machines_cache_from_dns call result in preliminary resolution of hostnames to ip addresses and passing of ip addresses to connect?